frontend/lib/src/components/widgets/DataFrame/DataFrame.tsx

```diff
@@ -768,7 +774,7 @@ function DataFrame({
           // Configure resize indicator to only show on the header:
           resizeIndicator={"header"}
           // Freeze all index columns:
-          freezeColumns={freezeColumns}
+          freezeColumns={numIndexColumns}
```

Comment:
`numIndexColumns` seems to be the [number of columns that are index columns](https://github.com/streamlit/streamlit/pull/8651/files#diff-eda1f0faafab9bbafe2c7b8b34241c35192488443ab3e02f0291d91e40ef632fR264). Why can this length be used to freeze the columns? Are they always the first X columns, so if we have 3 index columns we know that we have to freeze 0,1, and 2?

Reply:
ah yeah looks like this based also on [this logic here](https://github.com/streamlit/streamlit/pull/8651/files#diff-2db77815a7ec126b03c452196060952ab03314d15d149f56b979333758887c82R175). We should have a comment about this fact [here](https://github.com/streamlit/streamlit/pull/8651/files#diff-eda1f0faafab9bbafe2c7b8b34241c35192488443ab3e02f0291d91e40ef632fR262)

Reply:
Yep, index columns are always at the beginning. But I change this in the latest commit to actually pass in columns and loop through it. Its more explicit and future-proof since we also can use it to eventually add other features (e.g. making some columns unselectable based on column config).